conda/plugins/types.py

```diff
@@ -24,11 +24,23 @@ class CondaSubcommand(NamedTuple):
 
 class CondaVirtualPackage(NamedTuple):
     """
-    Conda subcommand entry.
+    A conda virtual package.
 
     :param name: Virtual package name (e.g., ``my_custom_os``).
     :param version: Virtual package version (e.g., ``1.2.3``).
     """
 
     name: str
     version: str | None
+
+
+class CondaSolver(NamedTuple):
+    """
+    A conda solver.
+
+    :param name: Solver name (e.g., ``custom-solver``).
+    :param backend: Callable that will be instantiated as the solver backend.
+    """
+
+    name: str
+    backend: Callable
```

Comment:
Shoudn't this type be a class of `conda.core.solve.Solver`? Or a `Protocol` implementing the API? A `Callable` is too broad imo. If we want to leave as is, then a pointer about the expected API contract (`__init__` args, public methods and their return types) should be added to the docs.

Reply:
Will using `conda.core.solve.Solver` here include subclasses of it or do we need to use `typing.Type[conda.core.solve.Solver]`?

Reply:
I remember now why this was a problem, our use of the plugin manager in the various parts of conda triggered a cyclic import error when importing from the `conda.core.solve` here. We could inline these imports there and see if that helps.

Reply:
I've moved the loading of the plugin manager back into the context object (as it used to be) in https://github.com/conda/conda/pull/11993/commits/36430f1ceacc441357917ff7e75cfbdb28994b41, so that I don't have to inline it all over the place.